exercises/bowling/example.rb

```diff
@@ -0,0 +1,105 @@
+class Game
+  class InvalidRoll < StandardError
+    def initialize(msg = "Roll must be between 0 and 10")
+      super
+    end
+  end
+
+  class FramePinsExceeded < StandardError
+    def initialize(msg = "Frame has at most 10 pins.")
+      super
+    end
+  end
+
+  class GameInProgress < StandardError; end
+  class GameHasEnded < StandardError; end
+
+  VERSION = 1
+  TOTAL_FRAMES = 10
+  MINIMUM_ROLL_SCORE = 0
+  MAXIMUM_ROLL_SCORE = 10
+
+  def initialize
+    @frame_scores = []
+    @frame_buffer = []
+  end
+
+  def roll(pins)
+    validate_roll(pins)
+    @frame_buffer << pins
+    adjust_frames
+  end
+
+  def score
+    raise GameInProgress unless @frame_scores.size == TOTAL_FRAMES
+    @frame_scores.reduce(:+)
+  end
+
+  private
+
+  def validate_roll(pins)
+    raise InvalidRoll if pins < MINIMUM_ROLL_SCORE
+    raise InvalidRoll if pins > MAXIMUM_ROLL_SCORE
+
+    if second_roll_in_frame?
+      raise FramePinsExceeded if @frame_buffer[0] + pins > MAXIMUM_ROLL_SCORE
+    end
+
+    raise GameHasEnded if @frame_scores.size == TOTAL_FRAMES
```

Comment:
This is the second time I have seen the `@frame_scores.size == TOTAL_FRAMES`so should this be named something like "game_over?"